File: EasyProblems/Java/twosum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        var numToIndex = new HashMap<Integer, Integer>();
        
        for(var i = 0; i < nums.length; i++) {
            if (numToIndex.containsKey(target - nums[i])) {
                return new int[]{numToIndex.get(target - nums[i]), i};
            }
            numToIndex.put(nums[i], i);
        }
        
        return new int[0];
    }
}

File: EasyProblems/JavaScript/twosum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var numToIndex = new Map();
    
    for(var i = 0; i < nums.length; i++) {
        if(numToIndex.has(target - nums[i])) {
            return [i, numToIndex.get(target - nums[i])];
        }
        
        numToIndex.set(nums[i], i);
    }
    
    return [];
};